catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CqlCommands.java

```diff
@@ -0,0 +1,225 @@
+
+        if (cqlFilter != null) {
+            return CQL.toFilter(cqlFilter);
+        }
+
+        final long start = getFilterStartTime();
+        final long end = filterCurrentTime;
+        final String temporalPropertyToFilter = getTemporalProperty();
+        final SimpleDateFormat formatter = new SimpleDateFormat(DATE_FORMAT);
+
+        if (StringUtils.isNotBlank(startDate) && StringUtils.isNotBlank(endDate)) {
+            return filterBuilder.attribute(temporalPropertyToFilter)
+                    .is()
+                    .during()
+                    .dates(formatter.parse(startDate), formatter.parse(endDate));
+        } else if (start > 0 && end > 0) {
+            return filterBuilder.attribute(temporalPropertyToFilter)
+                    .is()
+                    .during()
+                    .dates(new Date(start), new Date(end));
+        } else if (isUseTemporal) {
+            return filterBuilder.attribute(temporalPropertyToFilter)
+                    .is()
+                    .during()
+                    .last(start);
+        } else {
+            if (caseSensitive) {
+                return filterBuilder.attribute(Metacard.ANY_TEXT)
+                        .is()
+                        .like()
+                        .caseSensitiveText(searchPhrase);
+            } else {
+                return filterBuilder.attribute(Metacard.ANY_TEXT)
+                        .is()
+                        .like()
+                        .text(searchPhrase);
+            }
+        }
+    }
+
+    protected long getFilterStartTime() {
+        if (lastSeconds > 0) {
+            return filterCurrentTime - TimeUnit.SECONDS.toMillis(lastSeconds);
+        } else if (lastMinutes > 0) {
+            return filterCurrentTime - TimeUnit.MINUTES.toMillis(lastMinutes);
+        } else if (lastHours > 0) {
+            return filterCurrentTime - TimeUnit.HOURS.toMillis(lastHours);
+        } else if (lastDays > 0) {
+            return filterCurrentTime - TimeUnit.DAYS.toMillis(lastDays);
+        } else if (lastWeeks > 0) {
+            Calendar weeks = GregorianCalendar.getInstance();
+            weeks.setTimeInMillis(filterCurrentTime);
+            weeks.add(Calendar.WEEK_OF_YEAR, -1 * lastWeeks);
+            return weeks.getTimeInMillis();
+        } else if (lastMonths > 0) {
+            Calendar months = GregorianCalendar.getInstance();
+            months.setTimeInMillis(filterCurrentTime);
+            months.add(Calendar.MONTH, -1 * lastMonths);
+            return months.getTimeInMillis();
+        } else {
+            return 0;
+        }
+    }
+
+    protected String getTemporalProperty() {
+        if (StringUtils.isNotEmpty(temporalProperty)) {
+            Optional<String> property = Stream.of(TEMPORAL_PROPERTIES)
+                    .filter(temporalProperty::equalsIgnoreCase)
+                    .findFirst();
+            if (property.isPresent()) {
+                return property.get();
```

Comment:
or if temporal properties was a Map(name ->Attribute)  the entire method could be done something like
```java
Optional.ofNullable(temporalProperty)
    .filter(StringUtils::isNotEmpty)
    .map(TEMPORAL_PROPERTIES::get)
    .orElse(DEFAULT_TEMPORAL_PROPERTY)
```